packages/react-aria-components/src/Tabs.tsx

```diff
@@ -229,54 +232,43 @@ function TabInner({item, state}: {item: Node<object>, state: TabListState<object
   );
 }
 
-export interface TabPanelsProps<T> extends Omit<CollectionProps<T>, 'disabledKeys'> {}
-
-/**
- * TabPanels is used within Tabs as a container for the content of each tab.
- * The ids of the items within the <TabPanels> must match up with a corresponding item inside the <TabList>.
- */
-export function TabPanels<T extends object>(props: TabPanelsProps<T>) {
+function TabPanel(props: TabPanelProps, forwardedRef: ForwardedRef<HTMLDivElement>) {
   const {state} = useContext(InternalTabsContext)!;
-  let {portal, collection} = useCollection(props);
-  const selectedItem = state?.selectedKey != null
-    ? collection.getItem(state!.selectedKey)
-    : null;
-
-  return (
-    <>
-      {selectedItem && <SelectedTabPanel item={selectedItem} />}
-      {portal}
-    </>
-  );
-}
-
-/**
- * A TabPanel provides the content for a tab.
- */
-export function TabPanel(props: TabPanelProps): JSX.Element {
-  return Item(props);
-}
-
-function SelectedTabPanel({item}: {item: Node<object>}) {
-  const {state} = useContext(InternalTabsContext)!;
-  let ref = React.useRef<HTMLDivElement>(null);
-  let {tabPanelProps} = useTabPanel(item.props, state!, ref);
+  let ref = useObjectRef<HTMLDivElement>(forwardedRef);
+  let {tabPanelProps} = useTabPanel(props, state, ref);
   let {focusProps, isFocused, isFocusVisible} = useFocusRing();
 
+  let isSelected = state.selectedKey === props.id;
```

Comment:
should be in useTabPanel?
should the focusRing props be moved in there as well?

or maybe it should change if/when the RSP implementation matches this more closely?

Reply:
maybe, but I was trying to not touch the hooks too much for now.